### PR TITLE
Feature/616 unit test latest time series points

### PR DESCRIPTION
**Pull request by johevemi**
## Description

Make code in dataloader testable and create unit test for the two new functions

## References

* #616

### CODE REVIEW

File: source/databricks/geh_stream/shared/data_loader.py [hunk 1]
```diff
@@ -159,18 +160,24 @@ def load_time_series_points(args: Namespace, spark: SparkSession, metering_point
 
     df = filter_on_date(df, parse_period(args))
 
+    df = select_latest_point_data(df)
+
+    df = include_only_time_series_for_which_we_have_metering_point_master_data(df, metering_point_df)
```
Comment (BjarkeMeier) [inline source/databricks/geh_stream/shared/data_loader.py:165]:
```suggestion
    df = filter_time_series_by_metering_points(df, metering_point_df.select(col(Colname.metering_point_id))
```

File: source/databricks/tests/shared/test_data_loader.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    assert df.collect() == expected.collect()
+
+
+def test_include_only_time_series_for_which_we_have_metering_point_master_data(time_series_factory, metering_point_factory):
+    time_series_df_1 = time_series_factory(datetime(2020, 1, 1, 0, 0), metering_point_id="D01")
+    time_series_df_2 = time_series_factory(datetime(2020, 1, 2, 0, 0), metering_point_id="D02")
+    time_series_df = time_series_df_1.union(time_series_df_2)
+    metering_point_df = metering_point_factory(datetime(2020, 1, 1, 0, 0), datetime(2020, 1, 2, 0, 0), metering_point_id="D02")
+    df = include_only_time_series_for_which_we_have_metering_point_master_data(time_series_df, metering_point_df)
+    assert df.collect() == time_series_df_2.collect()
```
Comment (BjarkeMeier) [inline source/databricks/tests/shared/test_data_loader.py:39]: AAA comments and empty lines between would be nice.

**Review by BjarkeMeier — APPROVED** [review]

File: source/databricks/geh_stream/shared/data_loader.py [hunk 3]
```diff
@@ -159,18 +161,24 @@ def load_time_series_points(args: Namespace, spark: SparkSession, metering_point
...
 
+    df = select_latest_point_data(df)
+
+    df = filter_time_series_by_metering_points(df, metering_point_df.select(col(Colname.metering_point_id)))
+
+    return df
+
+
+def select_latest_point_data(df: DataFrame) -> DataFrame:
     # Select latest point data
```
Comment (lasrinnil) [inline source/databricks/geh_stream/shared/data_loader.py:172]: nit: Is this comment relevant?

**Review by lasrinnil — APPROVED** [review]

File: source/databricks/geh_stream/shared/data_loader.py (hunk 3 above)
Comment (johevemi) [reply source/databricks/geh_stream/shared/data_loader.py:172]: good point will remove it will remove it

**Pull request merged**